packages/components/src/DataList/DataList.utils.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { Children, ReactElement, isValidElement } from "react";
+import isEmpty from "lodash/isEmpty";
+import {
+  DataListHeader,
+  DataListItemType,
+  DataListItemTypeFromHeader,
+  DataListObject,
+} from "./DataList.types";
+import styles from "./DataList.css";
+import { EmptyState, EmptyStateProps } from "./components/EmptyState";
+import {
+  EMPTY_FILTER_RESULTS_ACTION_LABEL,
+  EMPTY_FILTER_RESULTS_MESSAGE,
+} from "./DataList.const";
+import { FormatDate } from "../FormatDate";
+import { InlineLabel } from "../InlineLabel";
+import { Text } from "../Text";
+
+/**
+ * Return the child component that matches the `type` provided
+ */
+export function getCompoundComponent<T>(
+  children: ReactElement | ReactElement[],
+  type: ReactElement<T>["type"],
+): ReactElement<T> | undefined {
+  const childrenArray = Children.toArray(children);
+  const element = childrenArray.find(
+    child => isValidElement<T>(child) && child.type === type,
+  );
+
+  // Comply with the return type without casting it
+  return isValidElement<T>(element) ? element : undefined;
+}
+
+/**
+ * Generate the default elements the DataList would use on the data provided.
+ */
+export function generateElementsFromData<T extends DataListObject>(data: T[]) {
+  type DataListElements = DataListItemType<typeof data>;
+
+  return data.map(item =>
+    Object.keys(item).reduce((acc, key: keyof DataListElements) => {
+      const currentItem = item[key];
+
+      if (!currentItem) {
+        return acc;
+      }
+
+      if (key === "tags" && Array.isArray(currentItem)) {
+        acc[key] = (
+          // TODO: Create a component specific for this with the experience we want
```

Comment:
Just confirming this todo is a later thing and not in this PR.

Reply:
yup. added a subtask for it in the same story